stat/test/wilcoxon_hyp.go

```diff
@@ -0,0 +1,190 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+func ensureDataConformance(x []float64, y []float64) {
+	if len(x) == 0 {
+		panic("x array length is zero")
```

Comment:
we don't call them arrays in Go (as this has a different meaning).
we call them slices.

also, `ensureDataConformance` isn't always called with `x` and `y` arguments so the panic message could be misleading.
(I think we also try to prefix panic messages with the package name: `"test: bla-bla"`)

Reply:
In general, we do these inline. They are more informative that way.